lib/Conversion/SCFToCalyx/SCFToCalyx.cpp

```diff
@@ -1313,6 +1347,21 @@ class BuildControl : public calyx::FuncOpPartialLoweringPattern {
                                          forLatchGroup.getName());
         if (res.failed())
           return res;
+      } else if (auto *callSchedPtr = std::get_if<CallScheduleable>(&group)) {
+        auto instanceOp = callSchedPtr->instanceOp;
+        OpBuilder::InsertionGuard g(rewriter);
+        auto callBody = rewriter.create<calyx::SeqOp>(instanceOp.getLoc());
+        rewriter.setInsertionPointToStart(callBody.getBodyBlock());
+        std::string initGroupName = "init_" + instanceOp.getSymName().str();
+        rewriter.create<calyx::EnableOp>(instanceOp.getLoc(), initGroupName);
+        SmallVector<Value, 4> instancePorts;
+        auto inputPorts = callSchedPtr->callOp.getOperands();
+        llvm::copy(instanceOp.getResults().take_front(inputPorts.size()),
```

Comment:
Can't this just be getResults(), or is there some type error?

Reply:
You shouldn't be able to do this here, because if you use `getResults()`, you will get all the ports,you will get unnecessary go, reset and other ports. However, here you only need the first few ports of the instance (equal to the number of results of `callOp`), which correspond to the parameters of `callOp`.

Reply:
Ah, ok.